packages/api/src/graphql/query.private.ts

```diff
@@ -290,6 +296,40 @@ export const GraphQLQuery = new GraphQLObjectType<undefined, Context>({
       }
     },
 
+    subscriptionAsCsv: {
+      type: GraphQLString,
+      args: {},
+      async resolve(root, args, {dbAdapter, authenticate}) {
+        const {roles} = authenticate()
+        authorise(CanGetUsers, roles)
+
+        const subscriptionsList: User[] = []
+
+        const getAllUsersWithSubscriptions = async ({endCursor}: any = {}) => {
+          const listResult = await dbAdapter.user.getUsers({
+            cursor: InputCursor(endCursor),
+            filter: {subscription: {}},
+            limit: Limit(1),
+            sort: UserSort.ModifiedAt,
+            order: SortOrder.Descending
+          })
+
+          subscriptionsList.push(...listResult.nodes)
+
+          if (listResult.totalCount > subscriptionsList.length) {
```

Comment:
`listResult` has the property `pageInfo` which contains the flag `hasNextPage`. I suggest you use that.